sd-rsyslog

```diff
@@ -74,15 +74,33 @@ def onInit():
     # emitted you must set 'level' to logging.DEBUG above.)
     logging.debug("onInit called")
 
-
     global process
     if not os.path.exists("/etc/sd-rsyslog.conf"):
         print("Please create the configuration file at /etc/sd-rsyslog.conf", file=sys.stderr)
         sys.exit(1)
     config = configparser.ConfigParser()
     config.read('/etc/sd-rsyslog.conf')
     logvmname = config['sd-rsyslog']['remotevm']
-    localvmname = config['sd-rsyslog'].get('localvm', gethostname())
+    localvmname = config['sd-rsyslog'].get('localvm', None)
+
+    # If no localvm name is specified, it must be supplied by Qubes OS. If this
+    # fails, we exit, to avoid falsely identified logs.
+    if localvmname is None:
+        try:
+            get_vm_name_process = Popen(["/usr/bin/qubesdb-read", "/name"],
+                                        stdout=PIPE, stderr=PIPE)
+            vm_name_output, vm_name_error = get_vm_name_process.communicate()
+            if vm_name_error != b"":
+                print("Error obtaining VM name via qubesdb-read:")
+                print(vm_name_error.decode("utf-8").strip())
```

Comment:
The `print()` statements here should really be `logging.error()` calls instead, no? Elsewhere in the file there's some inconsistency, but where `print()` is used it includes `file=sys.stderr`. In both cases, we should prefer `logging.error()` AFAICT.

Reply:
That makes more sense for sure. Will switch to `logging`.

Reply:
Done in  3afc03f776ee5df977a6715fbb46637fae737087, not tested yet.